recipes/opentracing-cpp/all/conanfile.py

```diff
@@ -0,0 +1,152 @@
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_MOCKTRACER"] = self.options.enable_mocktracer
+        self._cmake.definitions[
+            "BUILD_DYNAMIC_LOADING"
+        ] = self.options.enable_dynamic_load
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        self._cmake.definitions["BUILD_STATIC_LIBS"] = not self.options.shared
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.definitions["ENABLE_LINTING"] = False
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "OpenTracing"
+        self.cpp_info.names["cmake_find_package_multi"] = "OpenTracing"
+
+        if self.options.shared:
+            self.cpp_info.components["opentracing"].names[
+                "cmake_find_package"
+            ] = "opentracing"
+            self.cpp_info.components["opentracing"].names[
+                "cmake_find_package_multi"
+            ] = "opentracing"
+            self.cpp_info.components["opentracing"].libs = ["opentracing"]
+            self.cpp_info.components["opentracing"].bindirs = ["lib"]
+            if self.options.enable_dynamic_load and self.settings.os == "Linux":
+                self.cpp_info.components["opentracing"].system_libs.append("dl")
+        else:
+            self.cpp_info.components["opentracing-static"].names[
+                "cmake_find_package"
+            ] = "opentracing-static"
+            self.cpp_info.components["opentracing-static"].names[
+                "cmake_find_package_multi"
+            ] = "opentracing-static"
+            self.cpp_info.components["opentracing-static"].defines.append(
+                "OPENTRACING_STATIC"
+            )
+            if self.options.enable_dynamic_load and self.settings.os == "Linux":
+                self.cpp_info.components["opentracing-static"].system_libs.append("dl")
+
+            if self.settings.os != "Windows":
+                self.cpp_info.components["opentracing-static"].libs = ["opentracing"]
+            else:
+                self.cpp_info.components["opentracing-static"].libs = [
+                    "opentracing-static"
+                ]
```

Comment:
Please, avoid conditional conan components names. Stick to one conan component name, and conditionally set `.names[cmake_find_package]` etc

Reply:
Changed